.github/workflows/create-release.yml

```diff
@@ -0,0 +1,60 @@
+name: Create release
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Version for release"
+        required: true
+        type: string
+
+concurrency:
+  group: restrict-concurrency
+  cancel-in-progress: true
+
+permissions:
+  contents: write
+  pull-requests: write
+  id-token: write
+
+jobs:
+  create-release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: main
+
+      - name: Validate version
+        run: |
+          echo "Validating version"
+      #          TODO: check version matches semver
+      #          TODO: check version is greater than previous version
+
+      - name: Configure git
+        run: |
+          git config --global user.name 'Figstract release manager'
+          git config --global user.email 'figstractreleasemanager@anifichadia.com'
+
+      - name: Update version
+        run: |
+          NEW_VERSION="${{ github.event.inputs.version }}"
+
+          echo "Updating version to: $NEW_VERSION"
+          sed -i "s/^VERSION=.*$/VERSION=$NEW_VERSION/" gradle.properties
+
+          git add -A
+          git commit -m "Prepare release $NEW_VERSION"
+          git push
+
+      # Re-checkout to get the latest main
+      - uses: actions/checkout@v4
+        with:
+          ref: main
+
+      - name: Setup tag
+        run: |
+          git tag "${{ github.event.inputs.version }}"
+
+      - name: Create release
+        run: |
+          gh release create "${{ github.event.inputs.version }}" --verify-tag -t "${{ github.event.inputs.version }}" --generate-notes
```

Comment:
Drop the manual tag creation, turns out gh release create can do this (just omit the `--verify-tag`)

<img width="1150" alt="image" src="https://github.com/user-attachments/assets/08f6fa0f-14cd-45d1-85fc-efde528af29f" />

https://cli.github.com/manual/gh_release_create